### ISSUE TITLE
Does not add the negative sign to TA<0

### ISSUE THREAD

**Issue by joeygold** [microchip-pic-avr-solutions/mcp9808_arduino_driver#1]
**The negative sign needs to be added where  TA<0**

IE  multiply by -1

```
if ((UpperByte & 0x10) == 0x10){ //TA < 0°C
###     //Negative Temp
    UpperByte = UpperByte & 0x0F; //Clear SIGN
    Temperature = 256 - (((float)UpperByte * 16) + ((float)LowerByte / 16))*-1
    printf("\nNegative ");
}else //TA > 0°C
    Temperature = (((float)UpperByte * 16) + ((float)LowerByte / 16));
```

**Comment by M68426** [comment]
Hi @joeygold

Seems like the two statements have been wrongly labeled on our part. This conversion is based on the sample instruction code found in the [MCP9808 datasheet](https://ww1.microchip.com/downloads/en/DeviceDoc/25095A.pdf#page=25&zoom=auto,-205,488) (Example 5-1, page 25). Here the _if_ clause is covering TA < 0, while the _else_ clause covers TA >= 0.

Added a fix for this in [1.1.3](https://github.com/microchip-pic-avr-solutions/mcp9808_arduino_driver/releases/tag/1.1.3)
Here we changed the conversion equation for fahrenheit: `return ((temp * 1.8) + 32.0);`

This should cover negative values :)

**Comment by joeygold** [comment]
@M68426 
This needs a change also  line 94

Before
`temp = 256.0 - (upper_byte * 16.0 + lower_byte / 16.0);`

Change to 
`temp = (256.0 - (upper_byte * 16.0 + lower_byte / 16.0))*-1;`

**Comment by M68426** [comment]
Sorry about that. 
Fixed in [1.1.4](https://github.com/microchip-pic-avr-solutions/mcp9808_arduino_driver/releases/tag/1.1.4)

Thanks for reporting 👍
